dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/tests/Audacia.CodeAnalysis.Analyzers.Test/Rules/UseRecordTypesAnalyzerTests.cs

```diff
@@ -0,0 +1,322 @@
+using Audacia.CodeAnalysis.Analyzers.Rules.UseRecordTypes;
+using Audacia.CodeAnalysis.Analyzers.Settings;
+using Audacia.CodeAnalysis.Analyzers.Test.Base;
+using Audacia.CodeAnalysis.Analyzers.Test.Helpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Audacia.CodeAnalysis.Analyzers.Test.Rules;
+
+[TestClass]
+public class UseRecordTypesAnalyzerTests : CodeFixVerifier
```

Comment:
Do you think it would be worth adding any tests which have the declaration of `public record class PeopleDto(int Id, string Name)` or `public record struct Coordinates(double Y, double X)`?

Reply:
Gotta be honest I don't know what declaring a type as `record class` or `record struct` means? Would we want a diagnostic raised in those two examples?

Reply:
The main difference to know is `record struct` is a value type and `record class` is a reference type.

By default defining a record, it'll use the `record class` variation.

Reply:
Happy to leave as is and create an additional work item in the backlog for the additional unit test coverage.